docs/schema.md

```diff
@@ -0,0 +1,93 @@
+# Protect.js schema
+
+Protect.js lets you define a schema in TypeScript with properties that map to your database columns, and define indexes and casting for each column which are used when searching on encrypted data.
+
+## Creating schema files
+
+You can declare your Protect.js schema directly in TypeScript either in a single `schema.ts` file, or you can split your schema into multiple files. It's up to you.
+
+Example in a single file:
+
+```
+📦 <project root>
+ ├ 📂 src
+ │   ├ 📂 protect
+ │   │  └ 📜 schema.ts
+```
+
+or in multiple files:
+
+```
+📦 <project root>
+ ├ 📂 src
+ │   ├ 📂 protect
+ │   |   └ 📂 schemas
+ │   │     └ 📜 users.ts
+ │   │     └ 📜 posts.ts
+```
+
+## Understanding schema files
+
+A schema represents a mapping of your database, and which columns you want to encrypt and index. Thus, it's a collection of tables and columns represented with `csTable` and `csColumn`.
+
+The below is pseudo-code for how these mappings are defined:
+
+```ts
+import { csTable, csColumn } from "@cipherstash/protect";
+
+export const tableNameInTypeScript = csTable("tableNameInDatabase", {
+  columnNameInTypeScript: csColumn("columnNameInDatabase"),
+});
+```
+
+## Defining your schema
+
+Now that you understand how your schema is defined, let's dive into how you can configure your schema.
+
+Start by importing the `csTable` and `csColumn` functions from `@cipherstash/protect` and create a new table with a column.
+
+```ts
+import { csTable, csColumn } from "@cipherstash/protect";
+
+export const protectedUsers = csTable("users", {
+  email: csColumn("email"),
+});
+```
+
+**Searchable encryption**
+
+If you are looking to enable searchable encryption in a PostgreSQL database, you must declaratively enable the indexes in your schema by chanining the index options to the column.
+
+```ts
+import { csTable, csColumn } from "@cipherstash/protect";
+
+export const protectedUsers = csTable("users", {
+  email: csColumn("email").freeTextSearch().equality().orderAndSort(),
+});
+```
+
+## Available index options
+
+The following index options are available for your schema:
+
+| **Method** | **Description** | **SQL equivalent** |
+| ----------- | --------------- | ------------------ |
+| equality   | Enables a unique index for equality queries. | `WHERE email = 'example@example.com'` |
```

Comment:
It doesn't enable a unique index, just an exact one.
However, this can be used for uniqueness if required by adding a unique index on it at the database level.
We should link to the EQL docs for how to do that.

Reply:
I think that it's probably worth renaming the concept of a "unique index" in CipherStash to "exact" or "equality". "unique" is too easy to confuse with the Postgres concept of unique indexes and doesn't really capture that it's used for equality queries as well.